docs/asyncapi-file.md

```diff
@@ -3,4 +3,89 @@ title: "AsyncAPI Specification File"
+   
+``` mermaid
+graph LR
+    A[Template Context]
+    B{Generator}
+    C[Parser]
+    D[Render Engine]
+    E[AsyncAPI File] --> B
+  subgraph Generator Library
+    B -->| asyncapiString | C
+    C --> | asyncapi -> AsyncAPIDocument type | A
+    B--> | originalAsyncAPI -> Stringified document | A
+    A --> D
+  end
+  ```
+
+> :bulb: **Remember**
+> 
+> When you are writing your template, you get access to the specification file in two different forms; the originalAsyncAPI which is a stringified version of the specification `.yml` file, and the asyncapiDocument.
+
+> :memo: **Note:** 
+> Check out the [asyncapiDocument structure](https://github.com/asyncapi/parser-js/blob/master/API.md#module_@asyncapi/parser+AsyncAPIDocument)
+
+In the following sections, you will learn how to use either the **asyncapiString** or the **asyncapiDocument** in your template.
+
+#### Method 1: asyncapiString + template 
+The first way to generate a message-based API template using the generator tool is by using the asyncapiString. The asynapiString is a stringified version of the specification YAML or JSON file you pass as a parameter to the CLI `ag` command. During the generation process, the generator library calls `generator.originalAsyncAPI`, which returns a stringified version of the original spec file. This stringified version is what the generator library passes to the render engine, and the render engine makes it accessible in your output alias the template.
+
+**An example of an asyncapiString:**
+```
+const asyncapiString = `
+asyncapi: 2.4.0
+info:
+  title: Example to show stringified specification file
+  version: 1.0.0
+  `
+...```
+
+The generator library generates templates using the `generator.generateFromString` instance method, as shown in the sample code snippet below:
+
+```
+generator
+  .generateFromString(asyncapiString)
+  .then(() => {
+    console.log('Done!');
+  })
+  .catch(console.error);
+  ```
+
+#### Method 2: asyncapiDocument + template
+A major advantage of using the asyncApiDocument is that it simplifies the structure of the more complex spec file, enabling the developer to easily access the specification files' content by simply invoking a function. 
+
+Once the specification YAML or JSON file is passed as an input to the generator library, it is passed on to the [Parser](parser.md) library, which then manipulates the file to a more structured document called the asyncapiDocument. Once the parser returns the document to the generator, the generator passes it to the render engine. The render engine makes the asyncapiDocument accessible from your output alias, the template.
+
+From the template, you can access the document's content. For example, if you wished to extract the spec file version used in your ReadME, you can do that by running `asyncapiDocument.version()`. Or, to get the message payload, you can use the `asyncapiDocument.allMessages()` method.
```

Comment:
```suggestion
For example, if you want to extract the version of your API from AsyncAPI document, you can do that by calling `asyncapi.version()`. You can say that this one is easy to access from JSON objects, but there are more complex scenarios. For example, to get access to all messages from all channels, you can call `asyncapi.allMessages()` instead of iterating through a complex JSON object on your own.
```